app/database/operator/server_data_operator/transformers/category.ts

```diff
@@ -53,14 +53,14 @@ export const transformCategoryRecord = ({action, database, value}: TransformerAr
  * @param {RecordPair} operator.value
  * @returns {Promise<CategoryChannelModel>}
  */
-export const transformCategoryChannelRecord = ({action, database, value}: TransformerArgs): Promise<CategoryChannelModel> => {
-    const raw = value.raw as CategoryChannel;
-    const record = value.record as CategoryChannelModel;
+export const transformCategoryChannelRecord = ({action, database, value}: TransformerArgs<CategoryChannelModel, CategoryChannel>) => {
+    const raw = value.raw;
+    const record = value.record;
     const isCreateAction = action === OperationType.CREATE;
 
     // If isCreateAction is true, we will use the id (API response) from the RAW, else we shall use the existing record id from the database
     const fieldsMapper = (categoryChannel: CategoryChannelModel) => {
-        categoryChannel._raw.id = isCreateAction ? (raw?.id ?? categoryChannel.id) : record.id;
+        categoryChannel._raw.id = isCreateAction ? (raw?.id ?? categoryChannel.id) : record?.id || '';
```

Comment:
This could add an empty string as the id? What happens if that is the case?

Reply:
For this and the 0 case, the reality is that it should never happen. The record should have a value if we are not in `isCreateAction`. I can use the `!` assertion, but I though this would be less error prone. But open to suggestions.

Reply:
Yes I get it, perhaps the bang should work best in this case so that it throws?